hivemq-edge/src/frontend/src/extensions/datahub/designer/operation/OperationNode.utils.ts

```diff
@@ -0,0 +1,168 @@
+import { getConnectedEdges, getIncomers, Node } from 'reactflow'
+
+import { PolicyOperation, Schema, Script } from '@/api/__generated__'
+import {
+  DataHubNodeType,
+  DataPolicyData,
+  DryRunResults,
+  OperationData,
+  TransitionData,
+  WorkspaceState,
+} from '@datahub/types.ts'
+import { checkValidityJSScript } from '@datahub/designer/script/FunctionNode.utils.ts'
+import { checkValiditySchema } from '@datahub/designer/schema/SchemaNode.utils.ts'
+import { PolicyCheckErrors } from '@datahub/designer/validation.errors.ts'
+import { isFunctionNodeType, isSchemaNodeType } from '@datahub/utils/node.utils.ts'
+
+export function checkValidityTransformFunction(
+  operationNode: Node<OperationData>,
+  store: WorkspaceState
+): DryRunResults<PolicyOperation, Script | Schema>[] {
+  const { nodes, edges } = store
+
+  if (!operationNode.data.functionId || !operationNode.data.formData) {
+    return [
+      {
+        node: operationNode,
+        error: PolicyCheckErrors.notConfigured(operationNode, 'functionId, formData'),
+      },
+    ]
+  }
+
+  ///////// Check the function handle
+  const functions = getIncomers(operationNode, nodes, edges).filter(isFunctionNodeType)
+
+  if (!functions.length) {
+    return [
+      {
+        node: operationNode,
+        error: PolicyCheckErrors.notConnected(DataHubNodeType.FUNCTION, operationNode),
+      },
+    ]
+  }
+
+  ///////// Check the serializers
+  const serialisers = getIncomers(operationNode, nodes, edges).filter(isSchemaNodeType)
+
+  const connectedEdges = getConnectedEdges([...serialisers], edges).filter(
+    (e) => e.targetHandle === OperationData.Handle.SERIALISER || e.targetHandle === OperationData.Handle.DESERIALISER
```

Comment:
suggestion: quick consistency comment to align with the line down below 

```suggestion
    (edge) => edge.targetHandle === OperationData.Handle.SERIALISER || edge.targetHandle === OperationData.Handle.DESERIALISER
```